pkg/api/k8s_versions.go

```diff
@@ -64,206 +49,244 @@ const (
-		"dashboard":                      "kubernetes-dashboard-amd64:v1.6.3",
-		"addon-resizer":                  "addon-resizer:1.7",
-		"heapster":                       "heapster-amd64:v1.5.1",
-		"metrics-server":                 "metrics-server-amd64:v0.2.1",
-		"kube-dns":                       "k8s-dns-kube-dns-amd64:1.14.5",
-		common.AddonManagerComponentName: "kube-addon-manager-amd64:v8.6",
-		"dnsmasq":                        "k8s-dns-dnsmasq-nanny-amd64:1.14.5",
-		"rescheduler":                    "rescheduler:v0.3.1",
-	},
-	"1.6": {
-		"dashboard":                      "kubernetes-dashboard-amd64:v1.6.3",
-		"addon-resizer":                  "addon-resizer:1.7",
-		"heapster":                       "heapster-amd64:v1.3.0",
-		"metrics-server":                 "metrics-server-amd64:v0.2.1",
-		"kube-dns":                       "k8s-dns-kube-dns-amd64:1.14.5",
-		common.AddonManagerComponentName: "kube-addon-manager-amd64:v6.5",
-		"dnsmasq":                        "k8s-dns-dnsmasq-nanny-amd64:1.14.5",
-		"rescheduler":                    "rescheduler:v0.3.1",
+var kubernetesImageBaseDefaultImages = map[string]map[string]string{
+	common.KubernetesImageBaseTypeGCR: {
+		common.DashboardAddonName:              "kubernetes-dashboard-amd64:v1.10.1",
+		common.ExecHealthZComponentName:        "exechealthz-amd64:1.2",
+		common.HeapsterAddonName:               "heapster-amd64:v1.5.4",
+		common.CoreDNSAddonName:                "coredns:1.6.6",
+		common.KubeDNSAddonName:                "k8s-dns-kube-dns-amd64:1.15.4",
+		common.DNSMasqComponentName:            "k8s-dns-dnsmasq-nanny-amd64:1.15.4",
+		common.DNSSidecarComponentName:         "k8s-dns-sidecar-amd64:1.14.10",
+		common.TillerAddonName:                 "tiller:v2.13.1",
+		common.ReschedulerAddonName:            "rescheduler:v0.4.0",
+		common.ACIConnectorAddonName:           "virtual-kubelet:latest",
+		common.ContainerMonitoringAddonName:    "oms:ciprod01072020",
+		common.AzureCNINetworkMonitorAddonName: "networkmonitor:v0.0.6",
+		common.IPMASQAgentAddonName:            "ip-masq-agent-amd64:v2.5.0",
+		common.DNSAutoscalerAddonName:          "cluster-proportional-autoscaler-amd64:1.1.1",
 	},
 }
 
-// K8sComponentsByVersionMap represents Docker images used for Kubernetes components based on Kubernetes versions (major.minor.patch)
-var K8sComponentsByVersionMap map[string]map[string]string
+func getDefaultImage(image, kubernetesImageBaseType string) string {
+	return kubernetesImageBaseDefaultImages[kubernetesImageBaseType][image]
+}
 
-func init() {
-	K8sComponentsByVersionMap = getKubeConfigs()
+// kubernetesImageBaseVersionedImages is a convenience map for "kubernetesImageBase" image version references that are distinct across versions of Kubernetes
+// For example, cluster-autoscaler generally ships a per-Kubernetes-version build
+// The map supports GCR or MCR image string flavors
+var kubernetesImageBaseVersionedImages = map[string]map[string]map[string]string{
```

Comment:
These are also in the style of `kubernetesImageBase` + `<string suffix>`, but these image references are not static for all versions of k8s, hence the distinct version-value dictionary representation.